grobid-core/src/main/java/org/grobid/core/engines/FullTextParser.java

```diff
@@ -2032,16 +2029,35 @@ else if (label.equals("I-<figure>") || openFigure) {
     				}
     				int ind = row.lastIndexOf("\t");
 	    			figureBlock.append(row.substring(0, ind)).append("\n");
-	    		}
-    			else {
+	    		} else {
 	    			openFigure = false;
 	    		}
     			nb++;
-    		}
-    		else
+    		} else
     			openFigure = false;
     	}
 
+    	// If there still an open figure
+    	if (openFigure) {
+            while((tokenizationsFigure.size() > 0) &&
+                (tokenizationsFigure.get(0).getText().equals("\n") ||
+                    tokenizationsFigure.get(0).getText().equals(" ")) )
+                tokenizationsFigure.remove(0);
+
+            // process the "accumulated" figure
+            Pair<String,String> trainingData = parsers.getFigureParser()
+                .createTrainingData(tokenizationsFigure, figureBlock.toString(), "Fig" + nb);
+            if (trainingData!= null) {
```

Comment:
Might be worth extracting into a shared method?

Do you know under what circumstances `trainingData` would be `null`?